test/src/test/java/org/corfudb/integration/StreamingIT.java

```diff
@@ -73,27 +74,47 @@ public void loadProperties() {
      * A StreamListener implementation to be used in the tests.
      * This listener accumulates all updates streamed to it into a linked list that
      * can then be used to verify.
+     * This listener can throw an exception on 'n' number of updates, in order to
+     * trigger the onError(Throwable t) and consequently, automatically unsubscribe.
      */
     private class StreamListenerImpl implements StreamListener {
 
         @Getter
         private final String name;
 
+        private int updatesToError = 0;
+
+        @Getter
+        private Timestamp timestamp;
+
         @Getter
         private final LinkedList<CorfuStreamEntries> updates = new LinkedList<>();
 
+        private CountDownLatch latch = null;
+
         StreamListenerImpl(String name) {
             this.name = name;
         }
 
+        StreamListenerImpl(String name, int updatesToError, CountDownLatch latch) {
+            this(name);
+            this.updatesToError = updatesToError;
+            this.latch = latch;
+        }
+
         @Override
+        @SneakyThrows
```

Comment:
`StreamSubscriptionException` is an unchecked exception, is `SneakyThrows` required ?

Reply:
yeah, not needed. Initially I was throwing a raw exception and that's why I used this. Removed.